src/argilla/client/feedback/utils.py

```diff
@@ -0,0 +1,136 @@
+    """Generates a `pydantic.BaseModel` schema from a list of `FieldSchema` objects to validate
+    the fields of a `FeedbackDataset` object before inserting them.
+
+    Args:
+        fields: the list of `FieldSchema` objects to generate the schema from.
+        name: the name of the `pydantic.BaseModel` schema to generate. Defaults to "FieldsSchema".
+
+    Returns:
+        A `pydantic.BaseModel` schema to validate the fields of a `FeedbackDataset` object before
+        inserting them.
+
+    Raises:
+        ValueError: if one of the fields has an unsupported type.
+
+    Examples:
+        >>> from argilla.client.feedback.schemas import TextField
+        >>> from argilla.client.feedback.dataset import generate_pydantic_schema
+        >>> fields = [
+        ...     TextField(name="text", required=True),
+        ...     TextField(name="label", required=True),
+        ... ]
+        >>> FieldsSchema = generate_pydantic_schema(fields)
+        >>> FieldsSchema(text="Hello", label="World")
+        FieldsSchema(text='Hello', label='World')
+    """
+    fields_schema = {}
+    for field in fields:
+        if field.settings["type"] not in FIELD_TYPE_TO_PYTHON_TYPE.keys():
+            raise ValueError(
+                f"Field {field.name} has an unsupported type: {field.settings['type']}, for the moment only the"
+                f" following types are supported: {list(FIELD_TYPE_TO_PYTHON_TYPE.keys())}"
+            )
+        fields_schema.update(
+            {field.name: (FIELD_TYPE_TO_PYTHON_TYPE[field.settings["type"]], ... if field.required else None)}
+        )
+    return create_model(name, **fields_schema)
+
+
+def feedback_dataset_in_argilla(
+    name: Optional[str] = None,
+    *,
+    workspace: Optional[Union[str, rg.Workspace]] = None,
+    id: Optional[str] = None,
+) -> Tuple[bool, Optional["FeedbackDatasetModel"]]:
+    """Checks whether a `FeedbackDataset` exists in Argilla or not, based on the `name`, `id`, or the combination of
+    `name` and `workspace`.
+
+    Args:
+        name: the name of the `FeedbackDataset` in Argilla.
+        workspace: the name of the workspace in Argilla where the `FeedbackDataset` is located.
+        id: the Argilla ID of the `FeedbackDataset`.
+
+    Returns:
+        A tuple with a boolean indicating whether the `FeedbackDataset` exists in Argilla or not, and the `FeedbackDatasetModel`
+        object if the `FeedbackDataset` exists, `None` otherwise.
+
+    Raises:
+        ValueError: if the `workspace` is not a `rg.Workspace` instance or a string.
+        Exception: if the `FeedbackDataset` could not be listed from Argilla.
+
+    Examples:
+        >>> import argilla as rg
+        >>> rg.init(api_url="...", api_key="...")
+        >>> from argilla.client.feedback.dataset import feedback_dataset_in_argilla
+        >>> fds_exists, fds_cls = feedback_dataset_in_argilla(name="my-dataset")
+    """
+    assert name or (name and workspace) or id, (
```

Comment:
```suggestion
    assert (name and workspace) or name or id, (
```

Reply:
Otherwise, if `name` is provided, the condition `(name or workspace)` will never be evaluated